packages/jest-each/src/bind.ts

```diff
@@ -50,6 +58,35 @@ export default (cb: GlobalCallback, supportsDone: boolean = true) => (
     }
   };
 
+export const bindConcurrent = (
```

Comment:
why is this needed next to `bind`?

Reply:
Concurrent test is typed only to be a `() => Promise<any>` whereas a normal test is `() => Promise<any> | void | undefined`.
Can't think of a way to have a common type signature for the return type. Any suggestions?

Reply:
Hmm, ok. Could we perhaps add a generic argument to `bind` so it can have different return types?

Reply:
e.g. `bind<T extends GlobalCallback | GlobalCallbackConcurrent>(cb: T, supportsDone: boolean = true) : T` or something like that

Reply:
Resolved this with @mattphillips's approach